### PR TITLE
[code] use the value stored in the builder's Usage in AudioOutputStreamOpenSLES

### PR DESCRIPTION
**Pull request by pbodilis**
I guess this partly solves issue #168.

I'm not sure about the ContentType. Would that be for AudioInputStreamOpenSLES?

### CODE REVIEW

**Comment by pbodilis** [comment]
I signed it!

File: src/opensles/AudioOutputStreamOpenSLES.cpp [hunk 1]
```diff
@@ -28,6 +28,40 @@
...
+        case Usage::NotificationEvent:
+            openslStream = SL_ANDROID_STREAM_NOTIFICATION;
+            break;
+        case Usage::AssistanceAccessibility:
+        case Usage::AssistanceNavigationGuidance:
+        case Usage::AssistanceSonification:
+            openslStream = SL_ANDROID_STREAM_SYSTEM;
+            break;
+        case Usage::Game:
+            openslStream = SL_ANDROID_RECORDING_PRESET_VOICE_COMMUNICATION;
```
Comment (philburk) [inline src/opensles/AudioOutputStreamOpenSLES.cpp]: SL_ANDROID_RECORDING_PRESET_VOICE_COMMUNICATION is not a streamType.
No ideal match is available.
Maybe should be SL_ANDROID_STREAM_MEDIA.

File: src/opensles/AudioOutputStreamOpenSLES.cpp [hunk 2]
```diff
@@ -157,8 +191,19 @@ Result AudioOutputStreamOpenSLES::open() {
...
+        result = (*configItf)->SetConfiguration(configItf,
+                                                SL_ANDROID_KEY_STREAM_TYPE,
+                                                &presetValue,
+                                                sizeof(SLuint32));
+        if (SL_RESULT_SUCCESS != result) {
+            goto error;
+        }
     }
 
+
```
Comment (philburk) [inline src/opensles/AudioOutputStreamOpenSLES.cpp]: extra space

File: src/opensles/AudioOutputStreamOpenSLES.cpp (hunk 1 above)
Comment (pbodilis) [reply src/opensles/AudioOutputStreamOpenSLES.cpp]: Oh, right, my bad. Agreed on SL_ANDROID_STREAM_MEDIA.

**Comment by pbodilis** [comment]
I don't know about ContentType... I felt Usage was more in line. If ContentType were to be used, how would you prioritize one above the other? Or would you combine both of them? So far, I'm ok with ignoring the parameter as it is right now.

**Comment by dturner** [comment]
For me: run a test using dumpsys to verify that the stream types are set correctly

**Comment by pbodilis** [comment]
> For me: run a test using dumpsys to verify that the stream types are set correctly

Which part of dumpsys shows this?
My modification enabled me to change the right volume stream, so I guess it works fine.

**Comment by pbodilis** [comment]
I'm not sure my changes in my PR have been taken into account, am I missing something?

**Comment by dturner** [comment]
Sorry, that previous comment was literally for me. I need to run a dumpsys to verify that your changes operate as intended (which I'm sure they do), then I'll merge.

**Comment by dturner** [comment]
Steps to test: 

- Clone https://github.com/pbodilis/oboe
- Open `hello-oboe` project (in Android Studio 3.4 Canary 8)
- In `PlayAudioEngine::setupPlaybackStreamParameters` set the stream usage to each of the following `Media`, `VoiceCommunication`, `Alarm`, `Notification` and `Assistant`
- For each usage run `hello-oboe`, change audio API to OpenSL ES. 
- Open volume settings and verify that changing the volume for that stream type changes the volume of the app's audio output 

Note the mappings from usage to the volume sliders in the settings UI (in Android Pie) are as follows: 

- Media -> Media volume
- VoiceCommunication -> Call volume
- Alarm -> Alarm volume
- Notification -> Ring volume
- Assistant -> Ring volume

It'd be better if the above tests were run from the command line - @philburk any ideas how we could do this? Pretty sure it's not possible at present.

**Comment by dturner** [comment]
@pbodilis Thanks very much for doing this 👍

**Pull request merged**